File: Everything/Models/Budget/Expense.cs
using System;

namespace everything.Models
{
    public class Expense
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }

        public int ExpenseBudgetId { get; set; }
        public virtual ExpenseBudget ExpenseBudget { get; set; }
    }
}

File: Everything/ClientApp/src/models/budget/Expense.ts
export default class Expense {
    id: number = 0;
    name: string = '';
    date: Date | null = null;
    description: string = '';
    amount: number = 0;
    expenseBudgetId: number = 0;

    constructor(init?: Partial<Expense>) {
        Object.assign(this, init);
    }
}